### PR TITLE
Add SERIAL as node table primary key

### PR DESCRIPTION
**Pull request by ray6080**
This PR introduces [`SERIAL`](https://www.postgresql.org/docs/8.1/datatype.html#DATATYPE-SERIAL) as a special data type for primary key properties, whose values are same as internal node offsets. This can speed up data ingestion, as no index construction/lookup is required during data ingestion.
`SERIAL` is limited to primary key of node tables only for now.

This PR hasn't handled updates, index scan, and expression evaluations over `SERIAL`.

### CODE REVIEW

**Review by andyfengHKU — APPROVED** [review]

File: src/include/common/types/types.h [hunk 1]
```diff
@@ -53,6 +53,9 @@ KUZU_API enum DataTypeID : uint8_t {
     ANY = 0,
     NODE = 10,
     REL = 11,
+    // SERIAL is a special data type that is used to represent a sequence of INT64 values that are
+    // incremented by 1 starting from 0.
+    SERIAL = 12,
```
[the remark this answers is not in the document]
Comment (ray6080) [reply src/include/common/types/types.h:58]: See #1495

File: src/binder/bind/bind_ddl.cpp [hunk 2]
```diff
@@ -22,25 +22,37 @@ using namespace kuzu::catalog;
...
         throw BinderException("Node " + tableName + " already exists.");
     }
-    auto boundProperties = bindProperties(createNodeClause.getPropertyNameDataTypes());
+    auto boundProperties = bindProperties(createNodeTableClause.getPropertyNameDataTypes());
     auto primaryKeyIdx = bindPrimaryKey(
-        createNodeClause.getPKColName(), createNodeClause.getPropertyNameDataTypes());
+        createNodeTableClause.getPKColName(), createNodeTableClause.getPropertyNameDataTypes());
+    for (auto i = 0u; i < boundProperties.size(); ++i) {
+        if (boundProperties[i].dataType.typeID == SERIAL && primaryKeyIdx != i) {
+            throw BinderException("Serial property in node table must be the primary key.");
```
[the remark this answers is not in the document]
Comment (ray6080) [reply src/binder/bind/bind_ddl.cpp:37]: See #1496

File: src/storage/wal_replayer_utils.cpp [hunk 3]
```diff
@@ -39,25 +39,37 @@ void WALReplayerUtils::createEmptyDBFilesForNewRelTable(RelTableSchema* relTable
...
                 directory, nodeTableSchema->tableID, DBFileType::ORIGINAL),
             nodeTableSchema->getPrimaryKey().dataType);
         pkIndex->bulkReserve(0 /* numNodes */);
         pkIndex->flush();
+    } break;
+    case SERIAL: {
+        // DO NOTHING.
+    } break;
+    default: {
+        throw NotImplementedException("Only INT64 and STRING primary keys are supported");
```
Comment (tnachen) [inline src/storage/wal_replayer_utils.cpp:71]: And SERIAL too?

Comment (ray6080) [reply src/storage/wal_replayer_utils.cpp:71]: Hi @tnachen Nice catch! Thanks! Will fix this soon.

Comment (ray6080) [reply src/storage/wal_replayer_utils.cpp:71]: The patch is squeezed in #1540 .

**Pull request merged**